matplotlib3.3 shading="nearest" compatibility

johnomotani opened the pull request:
matplotlib-3.3 adds a new option `shading="nearest"` for `pcolormesh`. This PR makes `animatplot` compatible with that option. Now also works with `shading="gouraud"` when matplotlib=3.3. Don't know why `shading="gouraud"` still doesn't work with matplotlib<=3.2, I guess they changed something in the internals.

t-makaro commented on animatplot/blocks/image_like.py:
```diff
@@ -49,22 +49,39 @@ def __init__(self, *args, ax=None, t_axis=0, **kwargs):
-        Slice = self._make_pcolormesh_slice(i, 3)
-        self.quad.set_array(self.C[Slice].ravel())
+        if self.shading == "flat":
+            Slice = self._make_pcolormesh_flat_slice(i, 3)
```
As I said earlier, I think that this will cause problems if shading is flat but ```(Ny, Nx) != self.C.shape[Slice]```

t-makaro reviewed (changes requested):
That explains so much of the garbage that I had to do to get this to work in the first place. (Looking at the docs:

```
'flat': ... The dimensions of X and Y should be one greater than those of C; if they are the same as C, then a deprecation warning is raised, and the last row and column of C are dropped.
```
)

My understanding is that if the dimensions of X,Y match C, then it defaults to nearest, but if flat is specified then it will work as before where my ```_make_pcolormech_slice``` cuts off the last value).

The question that this leaves me with, is what happens if the shading is specified as flat (or auto), but the dimensions of X,Y is one greater than C as is it supposed to be? This was never something that I considered, but since matplotlib covers it, then I think that we should too.

**Please add logic and a test for when flat is specified and the dimensions of X,Y are 1 greater than C. (The logic should be as simple as changing the if statement where my second comment is).**

Thank you for this contribution.

johnomotani commented:
Thanks @t-makaro, you were right. I've fixed the case and added a test. Also added a test for `shading="auto"` which previously had a  bug in the condition that the tests missed.

ianhi commented:
> I guess they changed something in the internals.

I think you might be looking for this PR https://github.com/matplotlib/matplotlib/pull/16258

t-makaro commented:
I just made an 0.4.2 release to get this change out.

johnomotani commented:
👍 thanks @t-makaro!
